### ISSUE TITLE
"thread '<unnamed>' panicked" when transforming invalid SVG

### ISSUE THREAD

**Issue by Ysgorg** [lovell/sharp#1572]
For a particular SVG image which is invalid according to [The W3 markup validator](https://validator.w3.org/), I get the following error:

```
thread '<unnamed>' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:345:21
note: Run with `RUST_BACKTRACE=1` for a backtrace.
Aborted (core dumped)
```

While I'm not expecting that sharp can successfully transform invalid input, it would be nice if the error would be properly handled. 

To reproduce: 

```
const fs = require('fs');
const sharp = require('sharp');
const IMG_PATH = 'invalid.svg';
const transformer = sharp();
const imageStream = fs.createReadStream(IMG_PATH);
imageStream.pipe(transformer);
try {
  transformer.toBuffer(function(err, buffer, info){
    console.log("success");
  })
} catch (e) {
  console.log("Threw handlable error")
}
```

Link to [invalid.svg](https://svgshare.com/i/B5G.svg) and a [valid SVG](https://svgshare.com/i/B5R.svg) which does work (can't add SVGs to github comments).

I'm using sharp version 0.21.3

**Comment by lovell** [comment]
Hello, what platform/OS does this occur with? Are you using the prebuilt binaries? What does `console.log(sharp.versions)` display?

I tried the above example using sharp v0.21.3 on Ubuntu Linux and it displayed "success" (the `err` passed to the callback had the `Error: svgload: SVG rendering failed` message).

**Comment by Ysgorg** [comment]
Output of `console.log(sharp.versions)`: 

```
{ cairo: '1.15.12',
  croco: '0.6.12',
  exif: '0.6.21',
  expat: '2.2.6',
  ffi: '3.2.1',
  fontconfig: '2.12.6',
  freetype: '2.9.1',
  fribidi: '1.0.5',
  gdkpixbuf: '2.36.12',
  gettext: '0.19.8.1',
  gif: '5.1.4',
  glib: '2.56.1',
  gsf: '1.14.44',
  harfbuzz: '1.8.7',
  jpeg: '1.5.3',
  lcms: '2.9',
  orc: '0.4.28',
  pango: '1.42.3',
  pixman: '0.34.0',
  png: '1.6.34',
  svg: '2.43.4',
  tiff: '4.0.9',
  vips: '8.7.0',
  webp: '1.0.0',
  xml: '2.9.8',
  zlib: '1.2.11' }
```

I'm using Ubuntu 18.04. 

Installed sharp with npm, and I see this: `info sharp Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.7.0/libvips-8.7.0-linux-x64.tar.gz`. If I understand correctly these are prebuilt binaries downloaded in the first clause of the npm install script (`node install/libvips && node install/dll-copy && prebuild-install`). I also tried to run the second clause (`node-gyp rebuild && node install/dll-copy`), but I get the same error.

**Comment by lovell** [comment]
Thank you, that all looks OK. My best guess would be that you've run into an odd rust/librsvg bug but I'm unsure why I can't reproduce it on the same platform/versions.

Does the latest sharp work-in-progress branch fix things for you?

```sh
npm install lovell/sharp#uptake
```

(This relies on pre-compiled binaries that provide a more recent version of librsvg compiled using a more recent rustc.)

**Comment by Ysgorg** [comment]
Yes that does work. Looking forward to the release, thank you!

**Comment by kevinburkenotion** [comment]
Hi Lovell,
We are running into the same error with the same backtrace. Can you update or close this ticket if/when the more recent binaries make it into a tagged release?

Thanks!

**Comment by kevinburkenotion** [comment]
I can reproduce by sending a HTML page to `sharp` using the node:10.15.1-slim Docker container. I tried getting a backtrace but no useful info there:

```
' panicked at 'called `Option::unwrap()` on a `None` value', libcore/option.rs:345:21
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
Aborted
Error: Unexpected response code: 134, 0
```

I can't reproduce on the `#uptake` branch.

**Comment by kevinburkenotion** [comment]
@lovell - Is there a generalizable way to handle SIGABRT or Rust panics from Node?

**Comment by lovell** [comment]
sharp v0.22.0 now available, providing libvips v8.7.4 and librsvg v2.45.5.
